Sources/UIComponent/Components/StackComponent.swift

```diff
@@ -62,33 +70,88 @@ public protocol StackComponent: Component, BaseLayoutProtocol {
   var justifyContent: MainAxisAlignment { get }
   var alignItems: CrossAxisAlignment { get }
   var children: [Component] { get }
+  var wrapper : Wrapper { get }
 }
 
 extension StackComponent {
   public func layout(_ constraint: Constraint) -> Renderer {
-    var renderers = getRenderers(constraint)
-    let crossMax = renderers.reduce(CGFloat(0).clamp(cross(constraint.minSize), cross(constraint.maxSize))) {
-      max($0, cross($1.size))
+    
+    func frame(child: Renderer, primaryCross: CGFloat, primaryOffset: CGFloat) -> (CGPoint, CGSize) {
+         (point(main: primaryOffset, cross: primaryCross), child.size)
     }
-    if cross(constraint.maxSize) == .infinity, alignItems == .stretch {
-      // when using alignItem = .stretch, we need to relayout child to stretch its cross axis
-      renderers = getRenderers(Constraint(minSize: constraint.minSize,
-                                          maxSize: size(main: main(constraint.maxSize), cross: crossMax)))
+    func maxSize(_ l: (CGPoint, CGSize), _ r: (CGPoint, CGSize)) -> (CGPoint, CGSize) {
+        if cross(l.1) > cross(r.1) {
+            return l
+        } else {
+            return r
+        }
     }
+    
+    var renderers = getRenderers(constraint)
     let mainTotal = renderers.reduce(0) {
       $0 + main($1.size)
     }
     
-    let (offset, distributedSpacing) = LayoutHelper.distribute(justifyContent: justifyContent,
+    var (offset, distributedSpacing) = LayoutHelper.distribute(justifyContent: justifyContent,
```

Comment:
Let's not calculate offset and spacing here if `wrapper == .wrap` since they will not be used.